src/d/lyt/d_lyt_draw_mark.cpp

```diff
@@ -0,0 +1,595 @@
+    }
+
+    mAnm[DRAW_MARK_ANIM_IN].setFrame(0.0f);
+    mAnm[DRAW_MARK_ANIM_IN].setAnimEnable(true);
+    mAnm[DRAW_MARK_ANIM_RING_IN].setFrame(0.0f);
+    mAnm[DRAW_MARK_ANIM_RING_IN].setAnimEnable(true);
+
+    mLyt.calc();
+
+    mAnm[DRAW_MARK_ANIM_IN].setAnimEnable(false);
+    mAnm[DRAW_MARK_ANIM_RING_IN].setAnimEnable(false);
+
+    mStateMgr.changeState(StateID_In);
+
+    return true;
+}
+
+bool dLytDrawMark_c::remove() {
+    dPad::ex_c::getInstance()->setField_0x70();
+    mResAcc.detach();
+    return true;
+}
+
+bool dLytDrawMark_c::execute() {
+    dPad::ex_c::fn_80059220();
+    mStateMgr.executeState();
+
+    for (int i = 0; i < DRAW_MARK_NUM_ANIMS; i++) {
+        if (mAnmCtrlInUse[i]) {
+            if (mAnm[i].isEnabled()) {
+                mAnm[i].play();
+            }
+        }
+    }
+
+    mLyt.calc();
+    return true;
+}
+
+bool dLytDrawMark_c::draw() {
+    mLyt.addToDrawList();
+    return true;
+}
+
+void dLytDrawMark_c::initializeState_Wait() {}
+void dLytDrawMark_c::executeState_Wait() {}
+void dLytDrawMark_c::finalizeState_Wait() {}
+
+void dLytDrawMark_c::initializeState_In() {
+    mAnm[DRAW_MARK_ANIM_IN].setForwardOnce();
+    mAnm[DRAW_MARK_ANIM_IN].setToStart();
+    mAnm[DRAW_MARK_ANIM_IN].setAnimEnable(true);
+    dSndSmallEffectMgr_c::GetInstance()->playSound(SE_S_SEAL_APPEAR);
+    if (mVariant == 1) {
+        dSndSmallEffectMgr_c::GetInstance()->playSound(SE_S_SEAL_LINE_APPEAR_2);
+    } else if (mVariant == 2) {
+        dSndSmallEffectMgr_c::GetInstance()->playSound(SE_S_SEAL_LINE_APPEAR_3);
+    } else {
+        dSndSmallEffectMgr_c::GetInstance()->playSound(SE_S_SEAL_LINE_APPEAR_1);
+    }
+}
+void dLytDrawMark_c::executeState_In() {
+    if (mAnm[DRAW_MARK_ANIM_IN].isStop2()) {
+        mAnm[DRAW_MARK_ANIM_IN].setAnimEnable(false);
+        mStateMgr.changeState(StateID_RingIn);
+    }
+}
+void dLytDrawMark_c::finalizeState_In() {}
+
+void dLytDrawMark_c::initializeState_RingIn() {
+    mAnm[DRAW_MARK_ANIM_RING_IN].setForwardOnce();
+    mAnm[DRAW_MARK_ANIM_RING_IN].setToStart();
+    mAnm[DRAW_MARK_ANIM_RING_IN].setAnimEnable(true);
+    mAnm[DRAW_MARK_ANIM_BASE_LOOP].setAnimEnable(true);
+    mAnm[DRAW_MARK_ANIM_BASE_LOOP].setFrame(0.0f);
+}
+void dLytDrawMark_c::executeState_RingIn() {
+    if (mAnm[DRAW_MARK_ANIM_RING_IN].isStop2()) {
+        mAnm[DRAW_MARK_ANIM_RING_IN].setAnimEnable(false);
+        mNeedEnableLineLoop = true;
+        mStateMgr.changeState(StateID_MoveWait);
+    }
+}
+void dLytDrawMark_c::finalizeState_RingIn() {}
+
+void dLytDrawMark_c::initializeState_MoveWait() {
+    loadPanePos();
+    if (mNeedEnableLineLoop) {
+        mAnm[mLineLoopIndex + DRAW_MARK_ANIM_DRAW_LINE_LOOP_OFFSET].setAnimEnable(true);
+        mNeedEnableLineLoop = false;
+    }
+    field_0x874 = mStartPositions[mLineLoopIndex];
+    // unused
+    mVec2_c _ = mEndPositions[mLineLoopIndex] - mStartPositions[mLineLoopIndex];
+    if (field_0x8F4 == -1) {
```

Comment:
This variable seems to indicate the current line index. -1 indicating failure or a reset for the next line to draw?